src/SignhostAPIClient/Rest/SignHostApiClient.cs

```diff
@@ -26,6 +26,52 @@ public class SignHostApiClient
 		private readonly ISignHostApiClientSettings settings;
 		private readonly HttpClient client;
 
+		public static bool IsPostbackChecksumValid(
+			IDictionary<string, string[]> headers,
+			string body,
+			string sharedSecret,
+			out Transaction postbackTransaction)
+		{
+			postbackTransaction = null;
+			string[] postbackChecksumArray;
+			string postbackChecksum;
+			string calculatedChecksum;
+			Postback postback;
+
+			try {
+				postback = JsonConvert.DeserializeObject<Postback>(body);
+				postbackTransaction = postback;
+			} catch (InvalidCastException e) {
+				return false;
+			} catch {
+				return false;
+			}
+
+			if (headers.ContainsKey("Checksum")) {
+				headers.TryGetValue("Checksum", out postbackChecksumArray);
+				postbackChecksum = postbackChecksumArray.First();
+			} else {
+				postbackChecksum = postback.Checksum;
+			}
+
+			if (!string.IsNullOrEmpty(postbackTransaction.Id) &&
+				!string.IsNullOrEmpty(postbackChecksum)) {
```

Comment:
When placing an if statement across multiple lines it often is better readable to place the closing `) {` on a newline.
eg:
```
if (some
   thing
   too long
) {
  // bla bla
}
```